python/ray/_raylet.pyx

```diff
@@ -2264,6 +2264,14 @@ cdef execute_task_with_cancellation_handler(
                 f"Exited because worker reached max_calls={execution_info.max_calls}"
                 " for this method.")
 
+cdef void clean_up_gpu_object_callback(const CObjectID &c_object_id) nogil:
```

Comment:
Which thread does this callback run on? Can it get blocked by task execution on the main thread?

Reply:
TLDR: The callback will be executed on the IO thread.

1. The RPC server is running on `io_service_`.

https://github.com/ray-project/ray/blob/1bc0087403ca50538f7fa65af5fb5ce342d96b41/src/ray/core_worker/core_worker.cc#L524-L534

2. `io_service_` is running on the IO thread.

https://github.com/ray-project/ray/blob/1bc0087403ca50538f7fa65af5fb5ce342d96b41/src/ray/core_worker/core_worker.cc#L470